### PR TITLE
adding sign up end point

### LINKED ISSUE #69: add sign up end point
add sign up end point with post method and get queries

### PR DESCRIPTION
**Pull request by ebraheemabbas**
hash password and adding it to DB
return jwt object
closes #69

### CODE REVIEW

**Review by omriza5 — DISMISSED** [review]

**Review by HosenAgbaria — DISMISSED** [review]

**Review by Karyum — CHANGES_REQUESTED** [review]
im guessing you opened this branch from the login, it's a pain to review this so i will wait until the login is merged, and review this.

`make sure to update the master for this branch after the login is merged`

File: the-experts-BE/app.js
Comment (jamiecoe) [inline the-experts-BE/app.js]: Still dodgy formatting! It should be:

```js
const { comparePasswords, hashPassword } = require('./public/javascripts/passwordmangment');
```

**Review by HosenAgbaria — DISMISSED** [review]

File: the-experts-BE/app.js
Comment (jamiecoe) [inline the-experts-BE/app.js]: If you are putting the `jwt` in a cookie, why do you need to send it back in `res.json()`? Remember that, once set, cookies get sent back to server automatically with every request.

Also, sending `res.json(null)` is dangerous because your frontend will get errors if it tries to anything with the data.

Why not just set `jwt` cookie and send back helpful message to FE eg: `res.send('you are logged in successfully')`?

File: the-experts-BE/app.js
Comment (jamiecoe) [inline the-experts-BE/app.js]: No `let`s if the variable isn't being mutated!!

Also you could shorten this with ES6 spread syntax:

```js
const body = { ...req.body }
```

File: the-experts-BE/app.js
Comment (jamiecoe) [inline the-experts-BE/app.js]: What is `cookie` here? I can't see it being assigned anywhere?

Also you are just sending `jwt` back to FE, then carrying on with `hashPassword()`. Don't you want to check first of all if the user is already signed up, but checking if their username already exists in DB?

File: the-experts-BE/app.js
Comment (jamiecoe) [inline the-experts-BE/app.js]: If you are putting the jwt in a cookie, why do you need to send it back in res.json()? Remember that, once set, cookies get sent back to server automatically with every request.

Why not just set jwt cookie and send back helpful message to FE eg: res.send('you are logged in successfully')?

File: the-experts-fe/src/components/ReviewSubmit/ReviewSubmit.jsx
Comment (jamiecoe) [inline the-experts-fe/src/components/ReviewSubmit/ReviewSubmit.jsx]: Please no more stray `console.log()` 🙏🙏🙏

**Review by jamiecoe — CHANGES_REQUESTED** [review]

**Review by omriza5 — DISMISSED** [review]

**Review by omriza5 — DISMISSED** [review]

**Review by HosenAgbaria — DISMISSED** [review]

**Review by jamiecoe — APPROVED** [review]

**Review by omriza5 — APPROVED** [review]

**Review by HosenAgbaria — APPROVED** [review]

**Review by omriza5 — APPROVED** [review]

**Pull request merged**
